src/ducks/Watchlists/Actions/Edit/EditForm/Assets.js

```diff
@@ -0,0 +1,200 @@
+import React, { useState, useRef, useEffect, useMemo } from 'react'
+import cx from 'classnames'
+import Label from '@santiment-network/ui/Label'
+import Icon from '@santiment-network/ui/Icon'
+import Input from '@santiment-network/ui/Input'
+import Panel from '@santiment-network/ui/Panel'
+import { Checkbox } from '@santiment-network/ui/Checkboxes/Checkboxes'
+import { useTheme } from '../../../../../stores/ui/theme'
+import { useAllProjects, useOnClickOutside } from './hooks'
+import styles from './index.module.scss'
+import cardStyles from '../../../../../ducks/Watchlists/Widgets/Table/AssetCard.module.scss'
+import fieldStyles from '../../../../../ducks/Studio/Sidebar/ProjectSelector/index.module.scss'
+
+const VIEW_ITEM_COUNT = 4
+
+const Assets = ({ watchlist, onChange }) => {
+  const ref = useRef()
+  const [isSearchMode, setIsSearchMode] = useState(false)
+  const [filter, setFilter] = useState('')
+  const watchlistProjects = useMemo(
+    () => (watchlist ? watchlist.listItems.map(l => l.project) : []),
+    [watchlist]
+  )
+  const [filteredProjects, setFilteredProjects] = useState(watchlistProjects)
+  const watchListIDs = useMemo(
+    () =>
+      new Set(
+        filteredProjects.map(i => i.id),
+        [filteredProjects]
+      )
+  )
+  const [checkedItems, setCheckedItems] = useState(watchlistProjects)
+  const { isNightMode } = useTheme()
+  const { data } = useAllProjects(filter.toLowerCase())
+  const unusedProjects = useMemo(
+    () => data.filter(item => !watchListIDs.has(item.id)),
+    [data, watchListIDs]
+  )
+  const [showItems, setShowItems] = useState(false)
+
+  useEffect(() => {
+    const showTimeout = setTimeout(() => setShowItems(true), 300)
+    return () => clearTimeout(showTimeout)
+  }, [])
+
+  useOnClickOutside(ref, () => {
+    setFilter('')
+    setIsSearchMode(false)
+  })
+
+  useEffect(() => {
+    let items = watchlistProjects
+    if (filter && filter.length > 0) {
+      const lowercaseFilter = filter.toLowerCase()
+      const filterHelper = ({ name, ticker }) =>
+        name.toLowerCase().includes(lowercaseFilter) ||
+        ticker.toLowerCase().includes(lowercaseFilter)
+      items = items.filter(filterHelper)
+    }
+    setFilteredProjects(items)
+  }, [filter, watchlistProjects])
+
+  const checkboxClickHandler = (item, newValue) =>
+    setCheckedItems(old => {
+      let changed = false
+      let items = new Set(old)
+      const has = items.has(item)
+      if (!has && newValue) {
+        items.add(item)
+        changed = true
+      } else if (has && !newValue) {
+        items.delete(item)
+        changed = true
+      }
+      if (changed) {
+        items = Array.from(items)
+        if (onChange) onChange(items)
+        return items
+      }
+      return old
+    })
```

Comment:
- See the `[D-2-1]` 
- onChange can be trigger via `useEffect(() => onChange(watchlistProjects), [watchlistProjects, onChange])`